src/_stories/execute/helpers.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+from _stories.returned import Failure
+from _stories.returned import Result
+from _stories.returned import Skip
+from _stories.returned import Success
+
+
+def combine_parallel_outcomes(results):
+    result = None
+    if not results:
+        return result
+
+    kwargs = {}
+    for result in results:
+        if isinstance(result, (Failure, Result, Skip)):
+            return result
+
+        kwargs.update(result.kwargs)
+
+    result = Success(**kwargs)
```

Comment:
The problem with this approach is that it treats the whole parallel block as a single step of the story.

The execution of substories that runs in parallel should stop as soon as one of them returns Result, Failure, or throws an exception.

Such an approach makes this mostly impossible to implement.

Reply:
As we discussed (and there will be an issue about it), I disagree on the meaning of `Result` on multiple levels.
If you execute steps in parallel you expect that they will be executed.

If you'd like to introduce an option to abort on the first `Failure` or exception, that's probably doable with anyio but not with threads as there are no reliable ways to kill threads in the middle of the execution.

Furthermore, this creates nondeterministic failures since we may kill a parallel step in the middle of doing something important that cannot be rolled back.
If for example we're writing to a database which doesn't support transactions (or only optionally supports them such as newer versions of MongoDB) we may cause some kind of inconsistent state which is harder to fix.

I'd rather prevent that by returning multiple failure reasons if more than one of the steps failed or raised an Exception.

Reply:
When you'll propose changes to the semantics of the `Result` class, please take into consideration the `editors` project. We have to provide an easy way to migrate existed codebases to the new version of the library. We have enterprise users as I said earlier. See how much effort it took to migrate from `Success(foo='bar')` to the assignment expression.

Reply:
We will not have to stop threads if we don't execute substories as a single bulk of work in the thread. Loading steps separately to the worker will solve this.

Reply:
https://github.com/dry-python/stories/issues/88 have a solution to the recovery of the state of the system. It will be implemented in a compatible way with the `parallel` feature. Depending on what feature will be landed first.